diplomacy/test/src/LazyModuleSpec.scala

```diff
@@ -0,0 +1,444 @@
+
+      /**
+        * test getChildren function can only get child lazyModule,not grandchild
+        */
+      utest.assert(outerLM.getChildren == List(outerLM.InnerOtherLM, outerLM.innerLM))
+      utest.assert(outerLM.children == List(outerLM.InnerOtherLM, outerLM.innerLM))
+    }
+
+    test("test method childrenIterator and NodesIterator") {
+      implicit val p = Parameters.empty
+      val genOption = () => UInt(32.W)
+      class DemoSource(implicit valName: sourcecode.Name) extends BundleBridgeSource[UInt](Some(genOption))
+      class InSourceLazyModule extends LazyModule {
+        var flag = 0
+        val source = new DemoSource
+        source.makeSink()
+        lazy val module: LazyModuleImpLike = new LazyModuleImp(this) {
+          source.bundle := 1.U
+        }
+      }
+      class SourceLazyModule extends LazyModule {
+        val insource = LazyModule(new InSourceLazyModule)
+        val insourcesecond = LazyModule(new InSourceLazyModule)
+        val source = new DemoSource
+        var flag = 1
+        source.makeSink()
+        lazy val module = new LazyModuleImp(this) {
+          source.bundle := 4.U
+        }
+      }
+      var sourceNodeFlag = 0
+      var sinkNodeFlag = 0
+      val sourceModule = LazyModule(new SourceLazyModule)
+      chisel3.stage.ChiselStage.elaborate(sourceModule.module)
+      //test childrenIterator:childrenIterator return itself LazyModule name and Child LazyModule name
+      //Todo:why function childrenIterator return an empty type and always println in console
+      sourceModule.childrenIterator {
+        case lm: InSourceLazyModule =>
+          lm.flag += 1
+        case lm: SourceLazyModule =>
+          lm.flag += 1
+        case _ =>
+      }
+      utest.assert(sourceModule.insource.flag == 1)
+      utest.assert(sourceModule.insourcesecond.flag == 1)
+      utest.assert(sourceModule.flag == 2)
+      //test nodeIterator
+      //Todo:why node's name is <LazyModuleSpec$$$Lambda$1576> .
+
+      // FIXME:apply function nodeIterator will return this.nodes.foreach(iterfunc: BaseNode => Unit) twice
+      sourceModule.nodeIterator {
+        case basenode: BundleBridgeSource[UInt] => sourceNodeFlag += 1
+        case basenode: BundleBridgeSink[UInt]   => sinkNodeFlag += 1
+        case _ =>
+      }
+      utest.assert(sourceNodeFlag == 4)
+      utest.assert(sinkNodeFlag == 4)
+    }
+
+    test("test method line:Return source line that defines this instance") {
+      class DemoLazyModule(implicit p: Parameters) extends LazyModule {
+        lazy val module: LazyModuleImpLike = new LazyModuleImp(this) {}
+      }
+      implicit val p = Parameters.empty
+      // instance source line
+      val demoLM = LazyModule(new DemoLazyModule)
+      //test function <line> indicate the line of instantiating if lazyModule
+      //function line return a string,which line to define the LazyModule
+      utest.assert(demoLM.line.contains("158:30"))
```

Comment:
the `158:30` is tricky here, maybe need some other method to test the source line.